packages/pdfform/src/signPdf.ts

```diff
@@ -31,61 +26,36 @@ export async function signPdf(pdfStream: Readable, signature: PDFSignature[], tm
 	if (!signature || signature.length < 1) throw new Error('The signatureLocationArray is empty');
 
 	const tempFilePath = path.resolve(tmpFolderPath, randomFilename({id: 'temporary', ext: 'pdf'}));
-	// https://github.com/galkahana/HummusJS
-	const outStream = new PDFWStreamForFile(tempFilePath);
-	const inStream = await PDFRStreamForStream(pdfStream);
-	const pdfWriter = createWriterToModify(inStream, outStream);
-
-	let currentPage = 0;
-	let pageModifier: PDFPageModifier;
-	// I want to create an image for each signature and then draw them both to the stream.
-	const drawingArray = signature
-		.sort((a, b) => a.location.onPage - b.location.onPage)
-		.map((val, index) => async () => {
-			const {
-				signature: {data},
-				location: {width = 400, height = 100, x = 1, y = 1, onPage = 1},
-			} = val;
-			const isPageNumDifferent = currentPage.toString() !== onPage.toString();
-			const isNextPageNumDifferent = signature[index + 1] && signature[index + 1].location.onPage.toString() !== onPage.toString();
-
-			// if the page is not the same as the previous one then we want to create a new pageModifier.
-			if (isPageNumDifferent) {
-				pageModifier = new PDFPgMod(pdfWriter, onPage - 1, true);
-				currentPage = val.location.onPage;
-			}
-
-			if (!data) throw new Error('The signature data field is empty');
-
-			// create the signature image.
-			const signatureBuffer = Buffer.from(data.replace('data:image/png;base64,', ''), 'base64');
-			const signatureFilePath = path.resolve(tmpFolderPath, randomFilename({id: 'signature', ext: 'png'}));
-			await fs.promises.writeFile(signatureFilePath, signatureBuffer);
-
-			// I am giving it the height of 100 because the ratio is set to being proportional,
-			// which alleviates the reason for having a height prop.
-			const cxt = pageModifier.startContext().getContext();
-			cxt.drawImage(x, y, signatureFilePath, {
-				transformation: {width, height, proportional: true},
+	const writeStream = fs.createWriteStream(tempFilePath);
+	pdfStream.pipe(writeStream);
+
+	return new Promise(resolve => {
+		fs.readFile(tempFilePath, async (error, file) => {
+			if (error) throw error;
+
+			const pdfDoc = await PDFDocument.load(file);
+			const pages = pdfDoc.getPages();
+
+			await Promise.all(
```

Comment:
When using Promise.all there is no guarantee on the ordering of when the promises will be executed. Sometimes it could be half done executing a promise and start the next one and then switch back to the first. If ordering is important use @thx/promise-sequential. Otherwise Promise.all is fine.